src/mutation/updateItem.ts

```diff
@@ -0,0 +1,102 @@
+import { DocumentClient } from 'aws-sdk/clients/dynamodb';
+import { AnyObject, FilterOperators, TableConfig } from '../types';
+import { keyOperatorLookup } from '../query/queryBuilder';
+import { PromiseResult } from 'aws-sdk/lib/request';
+import { AWSError } from 'aws-sdk';
+
+interface ConditionExpressionItem {
+  key: string;
+  comparator: FilterOperators;
+  value: string | number | boolean | Array<number>;
+}
+
+interface AndOrExpression {
+  andOr: 'AND' | 'OR';
+}
+
+export type ConditionExpressionInput = Array<
+  ConditionExpressionItem | AndOrExpression
+>;
+
+interface ConditionExpressionReturn {
+  expression: DocumentClient.ConditionExpression;
+  attrValues: DocumentClient.ExpressionAttributeValueMap;
+}
+
+const buildConditionExpressions = (
+  conditionExpression: ConditionExpressionInput,
+): ConditionExpressionReturn => {
+  let expression = '';
+  const attrValues: DocumentClient.ExpressionAttributeValueMap = {};
+  for (let i = 0; i < conditionExpression.length; i++) {
+    const currentExpression = conditionExpression[i];
+    if ('andOr' in currentExpression) {
```

Comment:
May I suggest you to look into [Discriminated Unions](https://www.typescriptlang.org/docs/handbook/typescript-in-5-minutes-func.html#discriminated-unions) TypeScript feature to potentially make this cleaner and more scalable?